content/4_Gold/Combinatorics.mdx

```diff
@@ -102,7 +102,17 @@ because there is always exactly one way to construct an empty subset and a subse
 
 This recursive formula is commonly known as [Pascal's Triangle](https://brilliant.org/wiki/pascals-triangle/). 
 
-Additionally, we can optimize this recursion formula from $\mathcal{O}(2^n)$ to $\mathcal{O}(n^2)$ using [dynamic programming](/gold/intro-dp) (DP). You can learn more about how DP is used to optimize the binomial coefficient recursive formula [here](https://www.geeksforgeeks.org/binomial-coefficient-dp-9/).
+A naive implementation of this would use a recursive formula, like below:
+
+```cpp
+/** Computes nCr mod p using naive recurssion */
+int binomial(int n, int k, int p) {
+	if (k == 0 || k == n) return 1;
+	return (binomial(n - 1, k - 1, p) + binomial(n - 1, k, p)) % p;
+}
+```
+
+Additionally, we can optimize this recursion formula from $\mathcal{O}(2^n)$ to $\mathcal{O}(n^2)$ using [dynamic programming](/gold/intro-dp) (DP). We can cache the values of smaller binomials into a dp, and then call them in O(1) time when needed. The code below shows a bottom-up implementation of this type of DP.
```

Comment:
```suggestion
Additionally, we can optimize this from $\mathcal{O}(2^n)$ to $\mathcal{O}(n^2)$ using [dynamic programming](/gold/intro-dp) (DP) by caching the values of smaller binomials to prevent recalculating the same values over and over again. The code below shows a bottom-up implementation of this.